### DISCUSSION TITLE
creating a custom plot

### DISCUSSION THREAD

**Discussion by benefron** [bonsai-rx/bonsai#1189] [Q&A]
Hi,
I am running a behavioral task on Bonsai and I want to display the following:
1. I have a node (double) that records the the hit rate (number of hits/total number of hit+miss). I want to look at the running average of the last 10 trials and that it will update with each new trial for the latest 10. I couldn't find a way to do it.
2. I want to be able to see when the mouse licked the lick port for each trial. I have right now a subject reporting and documenting every timestamp when the mouse licked, but I want also some graphical representation of it to know how many times and when the mouse licked in each trial relative to the trial onset and the stimuli presentation.

Would love o hear any idea of how to do both,
Thanks,
Ben

**Comment by bruno-f-cruz** [comment]
Hey
1. You can use `Buffer(Reactive)` and apply an arbitrary workflow on each buffer using a `SelectMany`. Check the attached workflow for an example.

2. If you have the two events you want to use, I would probably use a `RollingGraph` and plot the two events (Trial event and Lick) with two distinct values. You could also just trigger the acquisition of one of the streams with another using a `WindowTrigger` or `BufferTrigger`.

Hope it helps
Bruno

[1189.zip](https://github.com/bonsai-rx/bonsai/files/10208809/1189.zip)
